src/Vault.sol

```diff
@@ -846,8 +863,8 @@ contract Vault {
     @dev All values expressed in the base currency of the vault with same number of decimals as the base currency.
     @return success Boolean indicating if there the margin position is successfully decreased.
      */
-    function decreaseMarginPosition(uint256 baseCurrency, uint256) public view onlyAuthorized returns (bool success) {
-        success = baseCurrency == vault._baseCurrency;
+    function decreaseMarginPosition(address baseCurrency, uint256) public view onlyAuthorized returns (bool success) {
```

Comment:
Should this view function have an auth modifier? Might be a hassle for bots/...

TBD if we should allow the `_setBaseCurrency` in the `increaseMaringPosition` as well, or just return false when the basecurrency isn't the same. Would be better if we can have both functions as public view functions without auth modifiers.

Reply:
Because in the future there should be more functions that can call this function